src/legacy/core_plugins/kibana_react/public/top_nav_menu/top_nav_menu_item.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import React, { MouseEvent } from 'react';
+import { EuiButtonEmpty } from '@elastic/eui';
+
+import { TopNavMenuData, TopNavMenuAction } from './top_nav_menu_data';
+
+interface Props {
+  data: TopNavMenuData;
+  onClick: (key: string, action: TopNavMenuAction, target?: any) => void;
+}
+
+export function TopNavMenuItem(props: Props) {
+  const menuData = props.data;
+
+  function isDisabled(): boolean {
+    const val =
+      typeof menuData.disableButton === 'function'
+        ? menuData.disableButton()
+        : menuData.disableButton;
+    return val || false;
```

Comment:
Done in a bit of a different way.
Instead of passing in a `data` object, I made `TopNavMenuItem` Props extend `TopNavMenuData`

```
interface Props extends TopNavMenuData {
  onClick: (key: string, action: TopNavMenuAction, target?: any) => void;
}
```

And then I could just use the react defaultProps:
```
TopNavMenuItem.defaultProps = {
  disableButton: false,
}
```